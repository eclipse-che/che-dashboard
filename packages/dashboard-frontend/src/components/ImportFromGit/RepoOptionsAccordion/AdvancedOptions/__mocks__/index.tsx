/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions';

export class AdvancedOptions extends React.PureComponent<Props> {
  public render() {
    const {
      containerImage,
      temporaryStorage,
      createNewIfExisting,
      memoryLimit,
      cpuLimit,
      onChange,
    } = this.props;

    return (
      <div>
        <div>Advanced Options</div>
        <div data-testid="advanced-options">{`${containerImage}, ${temporaryStorage}, ${createNewIfExisting}, ${memoryLimit}, ${cpuLimit}`}</div>
        <button
          onClick={() =>
            onChange('newContainerImage', !temporaryStorage, !createNewIfExisting, 1073741824, 1)
          }
        >
          Advanced Options Change
        </button>
      </div>
    );
  }
}
