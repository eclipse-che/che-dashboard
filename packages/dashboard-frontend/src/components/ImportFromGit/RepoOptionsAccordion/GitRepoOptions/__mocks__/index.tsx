/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/ImportFromGit/RepoOptionsAccordion/GitRepoOptions';

export class GitRepoOptions extends React.PureComponent<Props> {
  public render() {
    const { gitBranch, remotes, devfilePath, hasSupportedGitService, onChange } = this.props;

    return (
      <div>
        <div>Git Repo Options</div>
        <div data-testid="git-repo-options">{`${gitBranch}, ${JSON.stringify(remotes)}, ${devfilePath}, ${hasSupportedGitService}`}</div>
        <button
          onClick={() =>
            onChange(
              'newBranch',
              [{ name: 'test-updated', url: 'http://test' }],
              'newDevfilePath',
              true,
            )
          }
        >
          Git Repo Options Change
        </button>
      </div>
    );
  }
}
