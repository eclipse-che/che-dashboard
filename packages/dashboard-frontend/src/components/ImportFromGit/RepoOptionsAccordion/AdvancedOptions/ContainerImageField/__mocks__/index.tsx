/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions/ContainerImageField';

export class ContainerImageField extends React.PureComponent<Props> {
  public render() {
    const { containerImage, onChange } = this.props;

    return (
      <div>
        <div>Container Image</div>
        <div data-testid="container-image">{containerImage}</div>
        <button onClick={() => onChange('new-container-image')}>Container Image Change</button>
      </div>
    );
  }
}
