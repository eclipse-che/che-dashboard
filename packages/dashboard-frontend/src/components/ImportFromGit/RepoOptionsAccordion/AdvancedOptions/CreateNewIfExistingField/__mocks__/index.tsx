/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions/CreateNewIfExistingField';

export class CreateNewIfExistingField extends React.PureComponent<Props> {
  public render() {
    const { createNewIfExisting, onChange } = this.props;

    return (
      <div>
        <div>Create New If Existing</div>
        <div data-testid="create-new-if-existing">
          {createNewIfExisting !== undefined ? createNewIfExisting.toString() : 'undefined'}
        </div>
        <button onClick={() => onChange(!createNewIfExisting)}>
          Create New If Existing Change
        </button>
      </div>
    );
  }
}
