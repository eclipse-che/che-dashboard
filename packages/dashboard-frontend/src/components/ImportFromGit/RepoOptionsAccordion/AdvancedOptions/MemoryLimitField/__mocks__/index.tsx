/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions/MemoryLimitField';

export class MemoryLimitField extends React.PureComponent<Props> {
  public render() {
    const { memoryLimit, onChange } = this.props;

    return (
      <div>
        <div>Memory Limit</div>
        <div data-testid="memory-limit">{memoryLimit.toString()}</div>
        <button onClick={() => onChange(1073741824)}>Memory Limit Change</button>
      </div>
    );
  }
}
