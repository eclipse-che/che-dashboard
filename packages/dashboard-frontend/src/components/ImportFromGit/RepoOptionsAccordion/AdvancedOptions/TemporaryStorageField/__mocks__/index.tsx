/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/ImportFromGit/RepoOptionsAccordion/AdvancedOptions/TemporaryStorageField';

export class TemporaryStorageField extends React.PureComponent<Props> {
  public render() {
    const { isTemporary, onChange } = this.props;

    return (
      <div>
        <div>Temporary Storage</div>
        <div data-testid="temporary-storage">
          {isTemporary !== undefined ? isTemporary.toString() : 'undefined'}
        </div>
        <button onClick={() => onChange(!isTemporary)}>Temporary Storage Change</button>
      </div>
    );
  }
}
