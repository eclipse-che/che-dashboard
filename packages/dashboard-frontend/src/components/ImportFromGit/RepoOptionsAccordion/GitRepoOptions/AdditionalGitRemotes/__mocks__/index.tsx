/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/ImportFromGit/RepoOptionsAccordion/GitRepoOptions/AdditionalGitRemotes';

export class AdditionalGitRemotes extends React.PureComponent<Props> {
  public render() {
    const { remotes, onChange } = this.props;

    return (
      <div>
        <div>Git Remotes</div>
        <div data-testid="git-remotes">{remotes ? JSON.stringify(remotes) : 'undefined'}</div>
        <button onClick={() => onChange([{ name: 'test-updated', url: 'http://test' }], true)}>
          Git Remotes Change
        </button>
      </div>
    );
  }
}
