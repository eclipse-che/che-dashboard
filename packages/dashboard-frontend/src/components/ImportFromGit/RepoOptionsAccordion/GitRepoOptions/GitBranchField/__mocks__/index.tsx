/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/components/ImportFromGit/RepoOptionsAccordion/GitRepoOptions/GitBranchField';

export class GitBranchField extends React.PureComponent<Props> {
  public render() {
    const { gitBranch, onChange } = this.props;

    return (
      <div data-testid="git-branch-component">
        <div>Git Branch</div>
        <div data-testid="git-branch">{gitBranch}</div>
        <button onClick={() => onChange('new-branch')}>Git Branch Change</button>
      </div>
    );
  }
}
