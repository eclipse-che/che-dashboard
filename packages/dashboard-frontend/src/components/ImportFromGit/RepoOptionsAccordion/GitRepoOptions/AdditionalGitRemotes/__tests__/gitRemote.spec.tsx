/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { api } from '@eclipse-che/common';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { Provider } from 'react-redux';

import AdditionalGitRemote from '@/components/ImportFromGit/RepoOptionsAccordion/GitRepoOptions/AdditionalGitRemotes/gitRemote';
import { GitRemote } from '@/components/WorkspaceProgress/CreatingSteps/Apply/Devfile/getGitRemotes';
import getComponentRenderer, { screen } from '@/services/__mocks__/getComponentRenderer';
import { FakeStoreBuilder } from '@/store/__mocks__/storeBuilder';

const { createSnapshot, renderComponent } = getComponentRenderer(getComponent);

const mockOnChange = jest.fn();
const mockOnDelete = jest.fn();

const callbacks: { getValidation?: () => boolean } = {};

describe('AdditionalGitRemote', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('snapshot', () => {
    const snapshot = createSnapshot({ name: 'test-1', url: 'https://test-1.repo.git' });
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  describe('preset git remote', () => {
    describe('valid', () => {
      test('HTTPS remote', () => {
        renderComponent({ name: 'test-1', url: 'https://test-1.repo.git' });

        const inputName = screen.getByPlaceholderText('origin');
        expect(inputName).toHaveValue('test-1');

        const inputURL = screen.getByPlaceholderText('HTTP or SSH URL');
        expect(inputURL).toHaveValue('https://test-1.repo.git');

        const isValidation = callbacks.getValidation ? callbacks.getValidation() : false;
        expect(isValidation).toBe(true);
      });

      test('SSH remote', () => {
        renderComponent({ name: 'test-1', url: 'git@github.com:eclipse-che/che-dashboard.git' }, [
          { name: 'test', keyPub: 'publicKey' },
        ]);

        const inputName = screen.getByPlaceholderText('origin');
        expect(inputName).toHaveValue('test-1');

        const inputURL = screen.getByPlaceholderText('HTTP or SSH URL');
        expect(inputURL).toHaveValue('git@github.com:eclipse-che/che-dashboard.git');

        const isValidation = callbacks.getValidation ? callbacks.getValidation() : false;
        expect(isValidation).toBe(true);
      });
    });

    describe('invalid', () => {
      test('HTTPS remote', () => {
        renderComponent({ name: 'test-1', url: 'htps://test-1.repo.git' });

        const inputName = screen.getByPlaceholderText('origin');
        expect(inputName).toHaveValue('test-1');

        const inputURL = screen.getByPlaceholderText('HTTP or SSH URL');
        expect(inputURL).toHaveValue('htps://test-1.repo.git');

        const error = screen.queryByText('The URL or SSHLocation is not valid');
        expect(error).toBeDefined();

        const isValidation = callbacks.getValidation ? callbacks.getValidation() : true;
        expect(isValidation).toBe(false);
      });

      test('SSH remote', () => {
        renderComponent({ name: 'test-1', url: 'git@github.com:eclipse-che/che-dashboard.git' });

        const inputName = screen.getByPlaceholderText('origin');
        expect(inputName).toHaveValue('test-1');

        const inputURL = screen.getByPlaceholderText('HTTP or SSH URL');
        expect(inputURL).toHaveValue('git@github.com:eclipse-che/che-dashboard.git');

        const error = screen.queryByText('No SSH keys found');
        expect(error).toBeDefined();

        const isValidation = callbacks.getValidation ? callbacks.getValidation() : true;
        expect(isValidation).toBe(false);
      });
    });
  });

  test('delete the remote', async () => {
    renderComponent({ name: 'test-1', url: 'https://test-1.repo.git' });

    const inputName = screen.getByPlaceholderText('origin');
    expect(inputName).toHaveValue('test-1');

    const inputURL = screen.getByPlaceholderText('HTTP or SSH URL');
    expect(inputURL).toHaveValue('https://test-1.repo.git');

    const deleteButton = screen.getByTestId('remove-remote');
    await userEvent.click(deleteButton);

    expect(mockOnDelete).toHaveBeenCalledTimes(1);
  });

  test('git remote change', async () => {
    const remote = { name: 'test', url: 'https://test' };
    renderComponent(remote);

    const inputName = screen.getByPlaceholderText('origin');

    await userEvent.click(inputName);
    await userEvent.paste('-updated');

    expect(mockOnChange).toHaveBeenCalledTimes(1);
    expect(mockOnChange).toHaveBeenNthCalledWith(1, { name: 'test-updated', url: 'https://test' });

    const inputURL = screen.getByPlaceholderText('HTTP or SSH URL');

    await userEvent.click(inputURL);
    await userEvent.paste('-updated');

    expect(mockOnChange).toHaveBeenCalledTimes(2);
    expect(mockOnChange).toHaveBeenNthCalledWith(2, {
      name: 'test-updated',
      url: 'https://test-updated',
    });

    await userEvent.clear(inputURL);
    expect(mockOnChange).toHaveBeenNthCalledWith(3, { name: 'test-updated', url: '' });

    await userEvent.paste('https://test2');
    expect(mockOnChange).toHaveBeenNthCalledWith(4, { name: 'test-updated', url: 'https://test2' });
  });
});

function getComponent(remote: GitRemote, keys: api.SshKey[] = []) {
  const store = new FakeStoreBuilder().withSshKeys({ keys }).build();
  return (
    <Provider store={store}>
      <AdditionalGitRemote
        remote={remote}
        onChange={mockOnChange}
        onDelete={mockOnDelete}
        callbacks={callbacks}
      />
    </Provider>
  );
}
