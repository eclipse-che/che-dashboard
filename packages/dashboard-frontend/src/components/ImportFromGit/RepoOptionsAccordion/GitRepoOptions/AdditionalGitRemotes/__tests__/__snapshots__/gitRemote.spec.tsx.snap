// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalGitRemote snapshot 1`] = `
<div
  className="pf-l-flex"
>
  <div
    className="pf-m-align-right"
  >
    <div
      className="pf-c-form__group remoteName"
    >
      <div
        className="pf-c-form__group-label"
      >
        <label
          className="pf-c-form__label"
        >
          <span
            className="pf-c-form__label-text"
          >
            Remote Name
          </span>
          <span
            aria-hidden="true"
            className="pf-c-form__label-required"
          >
             
            *
          </span>
        </label>
         
      </div>
      <div
        className="pf-c-form__group-control"
      >
        <input
          aria-invalid={false}
          aria-label="Remote Name"
          className="pf-c-form-control pf-m-success"
          data-ouia-component-id="OUIA-Generated-TextInputBase-1"
          data-ouia-component-type="PF4/TextInput"
          data-ouia-safe={true}
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="origin"
          required={false}
          type="text"
          value="test-1"
        />
      </div>
    </div>
  </div>
  <div
    className="pf-m-grow pf-m-align-right"
  >
    <div
      className="pf-c-form__group remoteURL"
    >
      <div
        className="pf-c-form__group-label"
      >
        <label
          className="pf-c-form__label"
        >
          <span
            className="pf-c-form__label-text"
          >
            Remote URL
          </span>
          <span
            aria-hidden="true"
            className="pf-c-form__label-required"
          >
             
            *
          </span>
        </label>
         
      </div>
      <div
        className="pf-c-form__group-control"
      >
        <input
          aria-invalid={false}
          aria-label="Remote URL"
          className="pf-c-form-control pf-m-success"
          data-ouia-component-id="OUIA-Generated-TextInputBase-2"
          data-ouia-component-type="PF4/TextInput"
          data-ouia-safe={true}
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="HTTP or SSH URL"
          required={true}
          type="text"
          value="https://test-1.repo.git"
        />
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <div>
      <button
        aria-disabled={false}
        aria-label={null}
        className="pf-c-button pf-m-plain removeRemoteButton"
        data-ouia-component-id="OUIA-Generated-Button-plain-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe={true}
        data-testid="remove-remote"
        disabled={false}
        onClick={[Function]}
        role={null}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z"
          />
        </svg>
      </button>
      Remove Remote
    </div>
  </div>
</div>
`;
